src/mountfsd/mountwork.c

```diff
@@ -489,6 +489,18 @@ static int vl_method_mount_image(
                                 /* image_filter= */ NULL,
                                 dissect_flags,
                                 &di);
+                if (r == -ENOPKG && !(dissect_flags & DISSECT_IMAGE_NO_PARTITION_TABLE))
+                         /* Maybe it's a bare filesystem, try again. Note that this requires privileges, as
+                          * it is classified by the policy as an 'unprotected' image and will be refused
+                          * otherwise. */
```

Comment:
i don#t get this. why first invoke with NO_PARTITION_TABLE if you then immediately try again with it if it didn't find anything? why not pass it right-away then?

Reply:
Because otherwise the GPT logic is skipped entirely, that's how `dissect_image` works, this same fallback logic is used elsewhere as well for this reason

Reply:
uh? no? the logic that deals with no partition table cases just falls through if we don#t find just an fs?

Reply:
uh indeed, maybe it changed at some point? I was sure that's how it worked, or maybe I just hallucinated it. I'll fix it